src/shell/js/composeui-fdc3/src/ComposeUIDesktopAgent.ts

```diff
@@ -0,0 +1,312 @@
+                    resolve();
+                } catch(error) {
+                    reject(error);
+                }
+            } else {
+                this.currentChannel = channel;
+                resolve();
+            }
+        });
+    }
+
+    //TODO: should return AccessDenied error when a channel object is denied
+    //TODO: should return a CreationFailed error when a channel cannot be created or retrieved (channelId failure)
+    public getOrCreateChannel(channelId: string): Promise<Channel> {
+        // return new Promise<Channel>(async(resolve, reject) => {
+        //     let channel = this.appChannels.find(innerChannel => innerChannel.id == channelId);
+        //     if (!channel) {
+        //         try{
+        //             await this.invokeChannelCreationMessage(ComposeUITopic.getOrCreateChannel(), channelId, "app");
+        //         } catch(error) {
+        //             reject(error);
+        //         }
+        //         channel = new ComposeUIChannel(channelId, "app", this.messageRouterClient);
+        //         this.addChannel(channel);
+        //     }
+        //     resolve(channel);
+        // });
+        throw new Error("Not implemented.");
+    }
+
+    //TODO
+    public createPrivateChannel(): Promise<PrivateChannel> {
+        throw new Error("Not implemented");
+    }
+
+    public getCurrentChannel(): Promise<Channel | null> {
+        return Promise.resolve(this.currentChannel!);
+    }
+
+    //TODO: add messageRouter message that we are leaving the current channel to notify the backend.
+    public leaveCurrentChannel(): Promise<void> {
+        return new Promise<void>((resolve, reject) => {
+            this.currentChannel = undefined;
+            this.currentChannelListeners.forEach(async listener => {
+                const isUnsubscribed = listener.unsubscribe();
+                if (!isUnsubscribed) {
+                    reject(new Error(`Listener couldn't unsubscribe. IsSubscribed: ${isUnsubscribed}, Listener: ${listener}`));
+                }
+            });
+            this.currentChannelListeners = [];
+            resolve();
+        });
+    }
+
+    public getInfo(): Promise<ImplementationMetadata> {
+        return new Promise<ImplementationMetadata>((resolve) => {
+            const metadata = {
+                fdc3Version: "2.0.0",
+                provider: "ComposeUI",
+                providerVersion: "0.1.0-alpha.1", //TODO: version check
+                optionalFeatures: {
+                    OriginatingAppMetadata: false,
+                    UserChannelMembershipAPIs: false
+                }
+            } as ImplementationMetadata;
+            resolve(metadata);
+        });
+    }
+
+    //TODO
+    public getAppMetadata(app: AppIdentifier): Promise<AppMetadata> {
+        throw new Error("Not implemented");
+    }
+
+    //TODO
+    public getSystemChannels(): Promise<Channel[]> {
+        throw new Error("Not implemented");
```

Comment:
Return an empty array? That's a safer default implementation than throwing.

Reply:
Yeah it is a deprecated function which is an alias to the getUserChannels. I missed that, will update this:
https://github.com/finos/FDC3/blob/master/src/api/DesktopAgent.ts#L535-L537